app/assets/javascripts/discourse/app/adapters/topic-list.js

```diff
@@ -12,7 +12,13 @@ export function finderFor(filter, params) {
 
       for (const [key, value] of Object.entries(params)) {
         if (typeof value !== "undefined") {
-          urlSearchParams.set(key, value);
+          if (Array.isArray(value)) {
+            value.forEach((v) => {
+              urlSearchParams.append(`${key}[]`, v);
+            });
```

Comment:
This is actually a bug I discovered while working on this. Previously an array of `["tag", "tag2"]` will be represented as `?tags=tag,tag2` in the query string which will be parsed by Rails as `"tag,tag2"`. This change made it such that array values are properly set as `tags[]=tag&tags[]=tag2` which will actually be represented as an array on the server.